### PR TITLE
Implement SBUS output for external module bay.

### PR DESCRIPTION
**Pull request by schwabe**

### CODE REVIEW

**Comment by schwabe** [comment]
image
Current implementation how warn users of the voltage level when SBUS is selected.

**Comment by yanivasy** [comment]
By using the battery UART port as an another alternative for the SBUS output, its possible to get standard SBUS voltage levels.

If a beta version was already implemented,  I'll be happy to test it give a feedback.

**Comment by kilrah** [comment]
Since the pull request hasn't been merged yet you need to build yourself.

File: companion/src/firmwares/opentx/opentxeeprom.h [hunk 1]
```diff
@@ -98,15 +98,11 @@ class ProtocolsConversionTable: public ConversionTable
...
         addConversion(PULSES_DSMX, val++);
       }
-      if (IS_HORUS_OR_TARANIS(board)) {
+      if (IS_ARM(board)) {
         addConversion(PULSES_CROSSFIRE, val++);
         addConversion(PULSES_MULTIMODULE, val++);
         addConversion(PULSES_PXX_R9M, val++);
-      }
-      else if (IS_ARM(board)) {
-        val++;
```
Comment (bsongis) [inline companion/src/firmwares/opentx/opentxeeprom.h:107]: it was for removing CROSSFIRE support from SKY9X but perhaps we could add it soon

File: radio/src/opentx.h [hunk 2]
```diff
@@ -447,14 +444,20 @@ void memswap(void * a, void * b, uint8_t size);
...
 #endif
 #define IS_MODULE_R9M(idx)                (g_model.moduleData[idx].type==MODULE_TYPE_R9M)
 #define IS_MODULE_PXX(idx)                (IS_MODULE_XJT(idx) || IS_MODULE_R9M(idx))
 
+#if defined(DSM2)
+  #define IS_MODULE_DSM2(idx)             (idx==EXTERNAL_MODULE && g_model.moduleData[EXTERNAL_MODULE].type==MODULE_TYPE_DSM2)
+  #define IS_MODULE_SBUS(idx)             (idx==EXTERNAL_MODULE && g_model.moduleData[EXTERNAL_MODULE].type==MODULE_TYPE_SBUS)
+#else
+  #define IS_MODULE_DSM2(idx)             (false)
+#endif
```
Comment (bsongis) [inline radio/src/opentx.h:459]: no `IS_MODULE_SBUS()` here?

**Review by bsongis — APPROVED** [review]

File: companion/src/firmwares/opentx/opentxeeprom.h (hunk 1 above)
Comment (schwabe) [reply companion/src/firmwares/opentx/opentxeeprom.h:107]: Yeah, but it only does the mapping. If a protocol is actually availabe is on another place. And on the Sky9x and Taranis the eeprom ids are identical anyway, so I remove that distinction.

**Comment by yanivasy** [comment]
@bsongis , from what I see, your version in #2894 was using the S. Port pin  which later was used for CrossFire modules.
Correct me if I'm wrong but @schwabe new version uses the PPM pin.

**Comment by bsongis** [comment]
Crossfire and SBUS are not the same protocol, Crossfire is bidirectional

**Comment by yanivasy** [comment]
> Crossfire and SBUS are not the same protocol, Crossfire is bidirectional

@bsongis , I never claimed that (see my first comment on #5236). Just wanted to point out to difference in implementation of SBUS output with your version a couple of years ago and @schwabe's version now a days.

**Comment by yanivasy** [comment]
I'll be very grateful for a compiled build for Taranis+ X9D plus along with Multimodule (including SBUS) & sqt5font as build options for testing

**Comment by schwabe** [comment]
Just download the  nightly build it has sbus already in it

**Comment by yanivasy** [comment]
> Just download the nightly build it has sbus already in it

@schwabe , Starting from Nightly build companion-windows-2.2.1N391.exe ( thru 2.2.1N392 and including latest 2.2.1N393) there is  a problem with Companion updates.

A popup error with "**Unable to check for updates**" appears.

(Although I set "Use firmware nightly builds" & "Use Companion nightly builds" check box within the "Application Settings" tab).

Currently I was able to build only using 2.2.1N390 or an older version of Companion (for windows).
I double checked it with two different OS (win7 & win10).

**Comment by schwabe** [comment]
@yanivasy There is a problem with SSL in newer update url. Just install a firmware from your N390, the build server should still give you a current nightly

**Comment by yanivasy** [comment]
I've tested the sbus functionality with Taranis+and it working well. Thank's a lot.
Please include the SBUS output feature within 2.2.1.

**Comment by 3djc** [comment]
Since it is merged, it will be in 2.2.1

**Pull request merged**